datafusion/sql/src/unparser/expr.rs

```diff
@@ -504,6 +508,14 @@ impl Unparser<'_> {
             .collect::<Result<Vec<_>>>()
     }
 
+    pub(super) fn new_ident_quoted_if_needs(&self, ident: String) -> ast::Ident {
```

Comment:
Please add a method comments for a pub method

Reply:
I added some comments in 7a534fb.